README.md

```diff
@@ -192,7 +192,7 @@ The above command pulled the existing data out of Route53 and placed the results
 
 | Provider | Requirements | Record Support | Dynamic | Notes |
 |--|--|--|--|--|
-| [AzureProvider](/octodns/provider/azuredns.py) | azure-identity, azure-mgmt-dns | A, AAAA, CAA, CNAME, MX, NS, PTR, SRV, TXT | No | |
+| [AzureProvider](/octodns/provider/azuredns.py) | azure-identity, azure-mgmt-dns, azure-mgmt-trafficmanager | A, AAAA, CAA, CNAME, MX, NS, PTR, SRV, TXT | Yes (CNAMEs only) | |
```

Comment:
What does the provider do when a dynamic `A` or `AAAA` is defined? Given that dynamic is a yes/no and not a per type thing it seems like it'd probably just repeatedly show a change that can't be applied.

We may need to extend that to optionally support dynamic types if we want to ship this to prevent that.

Would it maybe make sense to do something like the following to call out that it's a little "non-standard" in terms of the intermediate CNAME? 

```suggestion
| [AzureProvider](/octodns/provider/azuredns.py) | azure-identity, azure-mgmt-dns, azure-mgmt-trafficmanager | A, AAAA, CAA, CNAME, MX, NS, PTR, SRV, TXT | Alpha (CNAMEs only) | |
```

We probably need to find a place to put the caveats you mention in https://github.com/octodns/octodns/pull/706#issue-642259778 so they're easily found if people go to look at using it.  There isn't a great place for that sort of thing ATM, so maybe just comments in the provider's docstring. As we break each provider out into it's own repo we'll have better options for that sort of thing.

Reply:
> it'd probably just repeatedly show a change that can't be applied

The provider throws an exception in `_extra_changes` if there's a non-CNAME dynamic record detected. This way we tell the user early on during the plan stage instead of giving up only later during the apply stage.

> extend that to optionally support dynamic types

not sure what this means or is required given above.

Agreed with your suggestion and about mentioning caveats in the doc-string. Commit coming up!

Reply:
The caveats text is too long for the doc-string, I've instead linked to this PR suggesting the reader to read it over.